README.md

```diff
@@ -6,8 +6,130 @@
 
+1. Handling of authentications and registration flows using suitable and safe authentication methods (HOTP, Ocra)
+2. Storing of user authentication and user secret data
+3. Sending of push notification (Google Cloud Messaging, Firebase Cloud Messaging, Apple Push Notifications and Cloud 
+   to Device Messaging)
+4. Storing of application state (for state persistency during registration and authentication workflows)
+
+# Who should use this library?
+Basically anyone who wants to implement a Tiqr server for their Tiqr Apps (Android & iOS). As Tiqr is tightly integrated
+into the OpenConext StepUp ecosystem, we chose to implement the Tiqr server into our [Tiqr GSSP](https://github.com/OpenConext/Stepup-tiqr)
+If you seek to implement a Tiqr backend yourself, you can look to that project for pointers. 
+
+# History
+A brief overview of notable points in time of this project
+
+- 2010: This project wat initially created in 2010. Zend framework 1 was chosen as the framework of choice.
+- 2012: The project was moved from a local SVN to GitHub
+- 2014: The UserSecretStorage and the Ocra implementation was created
+- 2015: GCM push message support was added, and several cleanup tasks where performed
+- 2019: FCM push message support was added
+- 2020: PHP 5 support was dropped
+- 2022: Unit & integration test coverage was added
+
+# Future strategy
+- Having a robust test coverage on the code should have a high priority on every new feature created or bug fixed.
+- Moving away from the long past Zend Framework is not a high priority, keeping the library working and increasing its 
+  predictability is far more important
+- Creating new code that is not tightly coupled ot the Zend Framework is encouraged
+
+# Using the library
+Examples on how to work with the library can be found in the aforementioned Tiqr GSSP. But some examples are included 
+here to get a basic understanding of the capabilities of the library.
+
+The API of the tiqr-server-libphp can be found in the `Tiqr` 'namespace' (`library/tiqr/Tiqr`). Notable classes found 
+here are:
+
+- `Autoloader` the ZF autoloader, used to load the following services and storage backends 
+- `Service` the main service class used to interact with the underlying features of the library
+- `Factories` the `UserStorage`, `UserSecretStorage` and `StateStorage` are creating the specified storage backend based 
+   on the way you configured Tiqr
+
+## Creating the Service
+An end to end example on how to configure, create and work with one of the libraries classes.
+
+### Config
+To create the Tiqr Service, we encourage you to create a factory in your consuming code base. In order to create the 
+service, you will first need to create a Tiqr server configuration. A brief example is listed below. Note that this
+example is taken from a Symfony parameters file. This YAML is later transformed to a PHP Array. But for readability the
+YAML format is retained in this example:
+
```

Comment:
This example is misleading IMO. The names of the keys and the structure of the configuration arrays for the Tiqr library classes are different and this file contains multiple configuration option arrays and uses a different structure.

The documentation for the configuration options for the Tiqr_service class is included in the classes itself: https://github.com/SURFnet/tiqr-server-libphp/blob/develop/library/tiqr/Tiqr/Service.php#L103-L158
The Tiqr_Service uses Tiqr_StateStorage and Tiqr_DeviceStorage (tokenexchange) factory classes. The configuration options and type to create are passed in the options to Tiqr_Service. For the configuration options of the concrete class, you need to look at the __constructor of the concrete class.
The Tiqr_UserStorage, Tiqr_UserSecretStorage, Tiqr_Message factory classes are used directly by the user of the library. For the configuration options you need to look at the __constructor of the concrete class.

For examples look at:
- Testcases: https://github.com/SURFnet/tiqr-server-libphp/blob/develop/library/tiqr/tests/Tiqr_ServiceTest.php#L9-L32 and https://github.com/SURFnet/tiqr-server-libphp/blob/develop/library/tiqr/tests/Tiqr_UserStorageTest.php
- The soon to be merged TestServer: https://github.com/SURFnet/tiqr-server-libphp/blob/feature/testserver/TestServer/TestServerController.php#L83-L123 and https://github.com/SURFnet/tiqr-server-libphp/blob/feature/testserver/TestServer/TestServerController.php#L125-L141

Reply:
Yes, I agree with you. Using the actual format is far more relevant in this context. Will update this.